src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_IndexerAccess.cs

```diff
@@ -105,20 +108,48 @@ private BoundExpression VisitIndexerAccess(BoundIndexerAccess node, bool isLeftO
             bool expanded,
             ImmutableArray<int> argsToParamsOpt,
             BitVector defaultArguments,
-            BoundIndexerAccess? oldNodeOpt,
+            BoundExpression oldNode,
             bool isLeftOfAssignment)
         {
+            Debug.Assert(oldNode is BoundIndexerAccess or BoundObjectInitializerMember);
+
             if (isLeftOfAssignment && indexer.RefKind == RefKind.None)
             {
                 TypeSymbol type = indexer.Type;
-                Debug.Assert(oldNodeOpt?.Type.Equals(type, TypeCompareKind.ConsiderEverything) != false);
-
-                // This is an indexer set access. We return a BoundIndexerAccess node here.
-                // This node will be rewritten with MakePropertyAssignment when rewriting the enclosing BoundAssignmentOperator.
+                Debug.Assert(oldNode.Type is not null);
+                Debug.Assert(oldNode.Type.Equals(type, TypeCompareKind.ConsiderEverything) != false);
```

Comment:
> != false [](http://example.com/codeflow?start=90&length=9)

It looks like the comparison is no longer necessary after the conditional logic is removed